horizon/facts/opal_forwarder.py

```diff
@@ -59,10 +64,26 @@ def create_data_source_entry(
     )
 
 
-def create_data_update_entry(entries: list[DataSourceEntry]) -> DataUpdate:
+def create_data_update_entry(
+    entries: list[DataSourceEntry],
+    *,
+    update_id: UUID | None = None,
+) -> DataUpdate:
     entries_text = ", ".join(entry.dst_path for entry in entries)
+    _update_id = (update_id or uuid4()).hex
```

Comment:
This has to be mandatory, otherwise we may accidentally make invalid data updates that contain different update ids...